src/main/kotlin/calculator/Number.kt

```diff
@@ -0,0 +1,48 @@
+package calculator
+
+import java.lang.NumberFormatException
+import java.util.Objects
+import java.util.Stack
+
+class Number private constructor(private val value: Int) : Word {
+
+    operator fun plus(other: Number) = Number(value + other.value)
+
+    operator fun minus(other: Number) = Number(value - other.value)
+
+    operator fun times(other: Number) = Number(value * other.value)
+
+    operator fun div(other: Number) = Number(value / other.value)
+
+    override fun equals(other: Any?): Boolean {
+        if (other is Number) return this.value == other.value
+        return false
+    }
+
+    override fun hashCode(): Int {
+        return Objects.hash(value)
+    }
+
+    override fun toString(): String {
+        return value.toString()
+    }
+
+    override fun calc(resultStack: Stack<Number>) {
+        resultStack.add(this)
+    }
+
+    companion object {
+        fun of(value: String): Number {
+            val v = value.trim()
+            if (v.isEmpty()) {
+                throw IllegalArgumentException("the value must not empty")
+            }
+
+            try {
+                return Number(v.toInt())
+            } catch (e: NumberFormatException) {
+                throw IllegalArgumentException("the value must number")
+            }
```

Comment:
```suggestion
           return Number(value.toIntOrNull() ?: throw IllegalArgumentException())
```
이렇게하면 한줄로 표현이 가능합니다 :)